core/src/main/java/hudson/cli/DeleteViewCommand.java

```diff
@@ -48,20 +52,48 @@ public String getShortDescription() {
     @Override
     protected int run() throws Exception {
 
-        view.checkPermission(View.DELETE);
+        boolean errorOccurred = false;
 
-        final ViewGroup group = view.getOwner();
-        if (!group.canDelete(view)) {
+        // Remove duplicates
+        final HashSet<String> hs = new HashSet<String>();
+        hs.addAll(views);
 
-            stderr.format("%s does not allow to delete '%s' view\n",
-                    group.getDisplayName(),
-                    view.getViewName()
-            );
-            return -1;
-        }
+        ViewOptionHandler voh = new ViewOptionHandler(null, null, null);
+
+        for(String view_s : hs) {
+            View view = null;
 
-        group.deleteView(view);;
+            try {
+                try {
+                    view = voh.getView(view_s);
+                    view.checkPermission(View.DELETE);
```

Comment:
I've used checkPermission() intentionally. hasPermission() returns True/False only, however checkPermission() provides useful details what went wrong (e.g. user had used a wrong credentials to authenticate himself)
I think it is better to provide detailed error message as much as we can.